modules/Products/Products.php

```diff
@@ -1353,32 +1353,42 @@ function unlinkRelationship($id, $return_module, $return_id) {
 		}
 	}
 
-	function save_related_module($module, $crmid, $with_module, $with_crmids, $otherParams = array()) {
-		$adb = PearDatabase::getInstance();
+    public function save_related_module($module, $crmid, $with_module, $with_crmids, $otherParams = array())
+    {
+        $adb = PearDatabase::getInstance();
+        $qtysList = array();
 
-		$qtysList = array();
-		if ($otherParams && is_array($otherParams['quantities'])) {
-			$qtysList = $otherParams['quantities'];
-		}
+        if ($otherParams && is_array($otherParams['quantities'])) {
+            $qtysList = $otherParams['quantities'];
+        }
 
-		if(!is_array($with_crmids)) $with_crmids = Array($with_crmids);
-		foreach($with_crmids as $with_crmid) {
-			$qty = 0;
-			if (array_key_exists($with_crmid, $qtysList)) {
-				$qty = (float) $qtysList[$with_crmid];
-			}
-			if (!$qty) {
-				$qty = 1;
-			}
+        if (!is_array($with_crmids)) {
+            $with_crmids = array($with_crmids);
+        }
 
-			if (in_array($with_module, array('Leads', 'Accounts', 'Contacts', 'Potentials', 'Products'))) {
-				$query = $adb->pquery("SELECT * FROM vtiger_seproductsrel WHERE crmid=? AND productid=?", array($crmid, $with_crmid));
-				if($adb->num_rows($query) == 0) {
-					$adb->pquery('INSERT INTO vtiger_seproductsrel VALUES (?,?,?,?)', array($with_crmid, $crmid, $with_module, $qty));
-				}
-			} else {
-				parent::save_related_module($module, $crmid, $with_module, $with_crmid);
-			}
-		}
-	}
+        foreach ($with_crmids as $with_crmid) {
+            $qty = 0;
```

Comment:
Instead of defining $qty = 0 and then two if statements, wouldn't something like `$qty = array_key_exists($with_crmid, $qtysList) ? (float)$qtysList[$with_crmid] : 1;` work better?

Reply:
$qty = !empty((float)$qtysList[$with_crmid]) ? (float)$qtysList[$with_crmid] : 1;
There is required value more than zero so !empty() should work fine